### ISSUE TITLE
`Vertex::build_bindings()` should be marked as `unsafe`

### ISSUE THREAD

**Issue by mamins1376** [glium/glium#1684]
The function is used to define memory layout of a vertex and since someone can provide incorrect layout information (e.g. refer to a field on struct that doesn't exist by returning out of bound offset).

Although it's normally implemented using `implement_vertex` macro, implementor can simply write her own.

**Comment by tomaka** [comment]
That's true. However it's the `Vertex` trait as a whole that should be marked as unsafe, because it's implementing the function that is unsafe, not calling it.

**Comment by mamins1376** [comment]
That's even better. changed the issue title.
